text/0000-ux-guidelines.md

```diff
@@ -0,0 +1,119 @@
+- Feature Name: ux-guidelines
+- Start Date: 2015-07-30
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+This is a conventions RFC to settle and document a number of UX guidelines.
+
+* Error Messages
+* Explain Messages
+* Compiler flags
+
+# Motivation
+
+Good user experience is paramount for happy users. We already have informal
+guidelines, and having them documented makes it easier for new developers to
+start contributing without regressing on user experience.
+
+# Detailed design
+
+This RFC includes a number of unrelated UX guidelines broken down into
+arbitrary subsections. It starts with an overall summary of the goals of
+these guidelines. Ideally, everything after this paragraph should be
+copyable into a `ux-guidelines.md` file into the rustc repo. This RFC is
+specifically not trying to create *new* guidelines - just codify the ones
+already exist informally.
+
+### Summary
+
+Don't forget the user. Whether human or another program, such as an IDE, a
+good user experience with the compiler goes a long way into making developer
+lives better. And when developer lives are better, they can make their users'
+lives better as well. We don't want users to be baffled by compiler output or
```

Comment:
I would rather have the "when developer lives are better" sentence removed. The first part does not necessarily follow the second.